core/Common.php

```diff
@@ -235,6 +235,24 @@ public static function mb_strtolower($string)
         return strtolower($string);
     }
 
+    /**
+     * Multi-byte strtoupper() - works with UTF-8.
+     *
+     * Calls `mb_strtoupper` if available and falls back to `strtoupper` if not.
+     *
+     * @param string $string
+     * @return string
+     * @api
+     */
+    public static function mb_strtoupper($string)
+    {
+        if (function_exists('mb_strtoupper')) {
+            return mb_strtoupper($string, 'UTF-8');
+        }
+
+        return strtoupper($string);
```

Comment:
here we should default to returning the original string, otherwise we risk destroying the unicode string when mbstring becomes un-available for some reason. Better keep original string than risking breaking it :+1:

Reply:
We aren't doing that in the method above for strtolower as well

Reply:
let's do it in all mb_ proxy methods to avoid destroying unicode strings

Reply:
Imho we shouldn't take care of that. As we can't avoid problems with unicode. `substr` for example might cut off within a multibyte char, `strlen` might return an incorrect length.
Also Twig for example does the same in some of their filters: if `mb_strtolower` isn't available they use `strtolower`

Reply:
>  As we can't avoid problems with unicode. 

Actually we can avoid potential problems by not touching the string... ideally we would detect when it's a unicode string but this may get tricky (or slow)

Reply:
Well ok, if you think it makes sense to change that now after it was [already over 6 years in the code](https://github.com/piwik/piwik/commit/a4287c9a1fadab7052c4fb50f7c9481785728c9b) and we didn't get any bug report about it...

Reply:
I think what @sgiehl was pointing out we might avoid it the problem there but get the problem somewhere else... not quite getting the problem as well @mattab ?